policy-controller/k8s/index/src/index.rs

```diff
@@ -189,7 +189,7 @@ impl Index {
 impl kubert::index::IndexNamespacedResource<k8s::Pod> for Index {
     fn apply(&mut self, pod: k8s::Pod) {
         let namespace = pod.namespace().unwrap();
-        let name = pod.name();
+        let name = pod.name_unchecked();
```

Comment:
looking at the docs, [`name_unchecked` panics if the resource's name is unset](https://docs.rs/kube/latest/kube/trait.ResourceExt.html#tymethod.name_unchecked). do we trust that this will always be set? the docs suggest preferring [`name_any`](https://docs.rs/kube/latest/kube/trait.ResourceExt.html#tymethod.name_any), which will not panic.